runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -0,0 +1,250 @@
+        public void reserveReflectObject(Object object);
+        public void reserveReflectMethod(ReflectMethod method);
+    }
+
+    public static XWalkCoreWrapper getInstance() {
+        return instance;
+    }
+
+    public static void initEmbeddedMode(XWalkCoreListener coreListener) {
+        reset(coreListener, null);
+        check();
+        init();
+    }
+
+    public static void reset(XWalkCoreListener coreListener, XWalkCoreWrapper coreWrapper) {
+        listener = coreListener;
+        instance = coreWrapper;
+        if (instance != null) instance.resetBridge();
+    }
+
+    public static void check() {
+        check(-1);
+    }
+
+    public static void check(int minAPIVersion) {
+        assert(instance == null);
+        provisionalInstance = new XWalkCoreWrapper(minAPIVersion);
+        if (listener == null) return;
+
+        XWalkCoreStatus status = provisionalInstance.mCoreStatus;
+        if (status == XWalkCoreStatus.MATCHED) {
+            listener.onXWalkCoreReady();
+        } else {
+            Throwable error = null;
+            if (status == XWalkCoreStatus.NOT_FOUND) {
+                error = new UnsatisfiedLinkError("XWalk Core Not Found");
+            } else {
+                error = new VerifyError("API Incompatible");
+            }
+            listener.onXWalkCoreStartupError(error, status);
+        }
+    }
+
+    public static void init() {
+        instance = provisionalInstance;
+        instance.initBridge();
+    }
+
+    public static boolean reserveReflectObject(Object object) {
+        if (instance != null || listener == null) return false;
+        listener.reserveReflectObject(object);
+        return true;
+    }
+
+    public static boolean reserveReflectMethod(ReflectMethod method) {
+        if (instance != null || listener == null) return false;
+        listener.reserveReflectMethod(method);
+        return true;
+    }
+
+    private XWalkCoreWrapper(int minAPIVersion) {
+        if (minAPIVersion <= XWALK_API_VERSION) {
+            mMinAPIVersion = minAPIVersion;
+        } else {
+            mMinAPIVersion = XWALK_API_VERSION;
+        }
+
+        mCoreStatus = findEmbeddedBridge();
+        if (mCoreStatus == XWalkCoreStatus.NOT_FOUND) {
+            mCoreStatus = findSharedBridge();
+        }
+    }
+
+    private XWalkCoreStatus findEmbeddedBridge() {
+        XWalkApplication xwalkApp = XWalkApplication.getApplication();
+        mBridgeLoader = XWalkCoreWrapper.class.getClassLoader();
+        Class<?> clazz = null;
+        Method method = null;
+        XWalkCoreStatus status = XWalkCoreStatus.NOT_FOUND;
+
+        try {
+            clazz = mBridgeLoader.loadClass(BRIDGE_PACKAGE + "." + XWALK_VIEW_DELEGATE);
+            method = clazz.getMethod("loadXWalkLibrary", Context.class);
+            method.invoke(null, xwalkApp);
+
+            status = XWalkCoreStatus.MATCHED;
+        } catch (UnsatisfiedLinkError | InvocationTargetException e) {
```

Comment:
UnsatiffiedLinkError means native library is built for ARM but running on IA image. There's no need to decompress and should return false directly.